src/GoogleCalendarFactory.php

```diff
@@ -19,14 +19,43 @@ public static function createForCalendarId(string $calendarId): GoogleCalendar
     }
 
     public static function createAuthenticatedGoogleClient(array $config): Google_Client
+    {
+        $authProfile = $config['default_auth_profile'];
+
+        switch ($authProfile) {
+            case 'service_account':
+                return self::createServiceAccountClient($config['auth_profiles']['service_account']);
+            case 'oauth':
+                return self::createOAuthClient($config['auth_profiles']['oauth']);
+        }
+
+        throw new \InvalidArgumentException("Unsupported authentication profile [{$authProfile}].");
```

Comment:
Create a dedicated exception for this.

Here's an example in another package of ours: https://github.com/spatie/laravel-backup/blob/85d6710ce77176ce926268a22825f9f70afb7aae/src/Exceptions/InvalidBackupDestination.php#L7